pkg/apis/pipeline/v1beta1/param_types_test.go

```diff
@@ -312,13 +312,16 @@ func TestNewResultReference(t *testing.T) {
 	}
 	for _, tt := range tests {
 		t.Run(tt.name, func(t *testing.T) {
-			got, err := v1beta1.NewResultRefs(tt.args.param)
-			if tt.wantErr != (err != nil) {
-				t.Errorf("TestNewResultReference/%s wantErr %v, error = %v", tt.name, tt.wantErr, err)
-				return
-			}
-			if d := cmp.Diff(tt.want, got); d != "" {
-				t.Errorf("TestNewResultReference/%s (-want, +got) = %v", tt.name, d)
+			expressions, ok := v1beta1.GetVarSubstitutionExpressionsForParam(tt.args.param)
+			if ok {
+				got, err := v1beta1.NewResultRefs(expressions)
+				if tt.wantErr != (err != nil) {
+					t.Errorf("TestNewResultReference/%s wantErr %v, error = %v", tt.name, tt.wantErr, err)
```

Comment:
Now that we have resultref.go we should also move these tests into resultref_test.go:

- TestNewResultReference
- TestHasResultReference
- TestLooksLikeResultRef

Reply:
ok, will do.

Reply:
If I change that, a new approval will be needed.